python/pyarrow/_parquet.pyx

```diff
@@ -1222,7 +1223,9 @@ cdef class ParquetReader(_Weakrefable):
 
         self.source = source
 
-        get_reader(source, use_memory_map, &rd_handle)
+        nf = get_native_file(source, use_memory_map)
+        (&rd_handle)[0] = <shared_ptr[CRandomAccessFile]>nf.get_random_access_file()
+        self.nf = nf
```

Comment:
Instead of basically doing here what `get_reader` does under the hood, could we modify `get_reader` to actually return the NativeFile object? (in other cases where it is being used, this return value will just be discarded) 
In that case we can here do something like:

```
self.nf = get_reader(source, use_memory_map, &rd_handle)
```

Reply:
Copy that. Initially I planned on doing this, but saw `get_native_file` and thought using that was better than changing the API of `get_reader`, but if that's no issue then I agree just returning it is easier.